tests/test_decision_making/test_acquisition_functions/test_thompson_sampling.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2023 The GPJax Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from beartype.typing import Callable
+from jax.config import config
+import jax.numpy as jnp
+import jax.random as jr
+from jaxtyping import (
+    Array,
+    Float,
+)
+import pytest
+
+from gpjax.dataset import Dataset
+from gpjax.decision_making.acquisition_functions.thompson_sampling import (
+    ThompsonSampling,
+)
+from gpjax.decision_making.utils import OBJECTIVE
+from gpjax.gps import (
+    ConjugatePosterior,
+    NonConjugatePosterior,
+    Prior,
+)
+from gpjax.kernels import RBF
+from gpjax.likelihoods import (
+    Gaussian,
+    Poisson,
+)
+from gpjax.mean_functions import Zero
+
+config.update("jax_enable_x64", True)
+
+
+def generate_dummy_1d_dataset() -> Dataset:
+    X = jr.uniform(key=jr.PRNGKey(42), minval=0.0, maxval=10.0, shape=(10, 1))
+    y = jnp.sin(X) + X + 1.0 + 10 * jr.normal(jr.PRNGKey(42), shape=X.shape)
+    D = Dataset(X, y)
+    return D
+
+
+def generate_dummy_2d_dataset() -> Dataset:
+    X = jr.uniform(key=jr.PRNGKey(42), minval=0.0, maxval=10.0, shape=(10, 2))
+    X0 = X[:, 0].reshape(-1, 1)
+    X1 = X[:, 1].reshape(-1, 1)
+    y = (
+        jnp.sin(X0)
+        + X1
+        + X0**2
+        + 1.0
+        + 10 * jr.normal(jr.PRNGKey(42), shape=X0.shape)
+    )
+    D = Dataset(X, y)
+    return D
+
+
+def generate_dummy_1d_test_X(num_test_points: int) -> Float[Array, "N 1"]:
+    test_X = jnp.linspace(0.0, 10.0, num_test_points).reshape(-1, 1)
+    return test_X
+
```

Comment:
this and the bits below are gonna be useful for testing all acquisition functions. Maybe pop in their own file. Perhaps @thomaspinder has a better idea? We could do something like the automatic way the kernels are tested

Reply:
Have refactored so that the standard functions I introduced for testing out the acquisition maximizer (e.g. `Forrester` etc.) can also be used here.